coramin/third_party/tests/test_minlplib_tools.py

```diff
@@ -0,0 +1,143 @@
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_nlincons=164319)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_nquadcons=139999)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_npolynomcons=13975)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_nsignomcons=801)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_ngennlcons=13975)
+        self.assertEqual(len(cases), 1704-2)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_njacobiannlnz=1623023)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_nlaghessiannz=1825419)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              max_nlaghessiandiagnz=100000)
+        self.assertEqual(len(cases), 1704-1)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              min_nnlsemi=1)
+        self.assertEqual(len(cases), 0)  # unit
+
+        cases = coramin.third_party.filter_minlplib_instances(instancedata_download_dir=os.path.join(current_dir, 'minlplib'),
+                                                              acceptable_objcurvature=['linear', 'convex'])
+        self.assertEqual(len(cases), 1330)  # regression
+        
+        os.remove(os.path.join(current_dir, 'minlplib', 'instancedata.csv'))
+        os.rmdir(os.path.join(current_dir, 'minlplib'))
+
+    def test_get_minlplib(self):
```

Comment:
Again, I wouldn't test this as part of the test harness, and instead test it as part of the overall test driver.